platform/backend/repository/memory/src/main/kotlin/io/hamal/repository/memory/record/Func.kt

```diff
@@ -130,6 +131,20 @@ class FuncMemoryRepository : RecordMemoryRepository<FuncId, FuncRecord, Func>(
 
     override fun list(query: FuncQuery): List<Func> = lock.withLock { FuncCurrentProjection.list(query) }
 
+    override fun listDeployments(funcId: FuncId): List<FuncDeploymentsRes> {
+        lock.withLock {
+            val recs = recordsOf(funcId)
+            return recs.map { rec ->
+                val dep = versionOf(funcId, rec.sequence())!!.deployment
+                FuncDeploymentsRes(
+                    message = dep.message,
+                    version = dep.version,
+                    deployedAt = rec.recordedAt()
+                )
+            }
+        }
```

Comment:
```suggestion
      lock.withLock {
            val recs = recordsOf(funcId)
            return recs.filterIsInstance<FuncDeployedRecord>().map { rec ->
                FuncDeploymentsRes(
                    message = rec.message,
                    version = rec.version,
                    deployedAt = rec.recordedAt()
                )
            }
        }
```

good idea - but we can read directly from the records here - this will lead that the inital deployment is not visible and I think thats fine --> probably you need to adjust some tests

Reply:
Same for sqlite